docs/adding_packages/conanfile_attributes.md

```diff
@@ -39,32 +39,26 @@ Where the SPDX guidelines do not apply, packages should do the following:
 
 ## Settings
 
-All recipes should list the four settings `os`, `arch`, `compiler` and `build_type` so Conan will compute a different package ID
-for each combination. There are some particular cases for this general rule:
+As a general rule, recipes should set the `settings` attribute to: `os`, `arch`, `compiler` and `build_type`, and let Conan compute the package ID based on the settings. Some exceptions apply, as detailed below. For cases not covered here, please reach out to the Conan Center maintainers team for assistance. The following list is not exhaustive:
 
-* **Recipes for _header only_ libraries** might omit the `settings` attribute, but in any case they should add
+* **Recipes for _header only_ libraries** or where the contents of the package are the same irrespective of settings, might omit the `settings` attribute altogether, unless there is any logic conditional on a setting value. If the recipe has options or dependencies, but the contents of the package are invariant irrespective of their values, the following logic must be added to ensure a single, unique package ID:
```

Comment:
I'm not sure it's a good idea. header_only recipes may need all settings indirectly when a recipe relies on a build system. It can be confusing for contributors.

Reply:
I believe that is covered by this part:

>  unless there is any logic conditional on a setting value

Could perhaps clarify further that this situation may be caused by some tools. The language is intentionally non-prescriptive: if a recipe needs the settings, then it should have those settings.

Reply:
that's hard to guess for contributors if any particular tool you import relies on some settings. it requires inspecting conan source code. even if it doesn't requires setting at the moment, it can potentially start in any new release (as all tools now have an access to conanfile, they may read settings).